google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,367 @@
+        managerAccountsToSearch.remove(customerId);
+        SearchPagedResponse response = null;
+        try {
+          // Issues a search request.
+          response =
+              googleAdsServiceClient.search(
+                  SearchGoogleAdsRequest.newBuilder()
+                      .setQuery(query)
+                      .setCustomerId(Long.toString(customerId))
+                      .build());
+
+          // Iterates over all rows in all pages to get all customer clients under the specified
+          // customer's hierarchy.
+          for (GoogleAdsRow googleAdsRow : response.iterateAll()) {
+            CustomerClient customerClient = googleAdsRow.getCustomerClient();
+
+            if (customerClient.getId().getValue() == seedCustomerId) {
+              rootCustomerClient = customerClient;
+            }
+
+            // The steps below map parent and children accounts. Continue here so that managers
+            // accounts exclude themselves from the list of their children accounts.
+            if (customerClient.getId().getValue() == customerId) {
+              continue;
+            }
+
+            // For all level-1 (direct child) accounts that are manager accounts, the above
+            // query will be run against them to create a map of managers to their
+            // child accounts for printing the hierarchy afterwards.
+            customerIdsToChildAccounts.put(customerId, customerClient);
+            // Checks if the child account is a manager itself so that it can later be processed
+            // and added to the map if it hasn't been already.
+            if (customerClient.getManager().getValue()) {
+              // A customer can be managed by multiple managers, so to prevent visiting the same
+              // customer multiple times, we need to check if it's already in the map.
+              boolean alreadyVisited =
+                  customerIdsToChildAccounts.containsKey(customerClient.getId().getValue());
+              if (!alreadyVisited && customerClient.getLevel().getValue() == 1) {
+                managerAccountsToSearch.add(customerClient.getId().getValue());
+              }
+            }
+          }
+        } catch (GoogleAdsException gae) {
+          System.out.printf(
+              "Unable to retrieve hierarchy for customer ID %d: %s%n",
+              customerId, gae.getGoogleAdsFailure().getErrors(0).getMessage());
+        }
+      }
+
+      // The rootCustomerClient will be null if the account hierarchy was unable to be retrieved
+      // (e.g. the account is a test account or a client account with an incomplete billing setup.
+      // This method returns null in these cases to add the seedCustomerId to the list of
+      // customer IDs for which the account hierarchy could not be retrieved.
+      if (rootCustomerClient == null) {
+        return null;
+      }
+
+      Map<CustomerClient, Multimap<Long, CustomerClient>> CustomerClientToHierarchy =
```

Comment:
Why is this variable named using a capital letter?

Reply:
Typo: fixed.